tests/Aql/LimitExecutorTest.cpp

```diff
@@ -0,0 +1,258 @@
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2018 ArangoDB GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Heiko Kernbach
+////////////////////////////////////////////////////////////////////////////////
+
+#include "BlockFetcherHelper.h"
+#include "catch.hpp"
+
+#include "Aql/AqlItemBlock.h"
+#include "Aql/AqlItemBlockShell.h"
+#include "Aql/ExecutorInfos.h"
+#include "Aql/InputAqlItemRow.h"
+#include "Aql/LimitExecutor.h"
+#include "Aql/ResourceUsage.h"
+#include "Aql/SingleRowFetcher.h"
+
+#include <velocypack/Builder.h>
+#include <velocypack/velocypack-aliases.h>
+
+using namespace arangodb;
+using namespace arangodb::aql;
+
+namespace arangodb {
+namespace tests {
+namespace aql {
+
+SCENARIO("LimitExecutor", "[AQL][EXECUTOR]") {
+  ExecutionState state;
+
+  ResourceMonitor monitor;
+  AqlItemBlockManager itemBlockManager(&monitor);
+  auto block = std::make_unique<AqlItemBlock>(&monitor, 1000, 1);
+  auto outputRegisters = std::make_shared<const std::unordered_set<RegisterId>>(0);
+  auto& registersToKeep = outputRegisters;
+  auto outputBlockShell =
+      std::make_unique<OutputAqlItemBlockShell>(itemBlockManager, std::move(block),
+                                                outputRegisters, registersToKeep);
+
+  // Special parameters:
+  // 4th offset
+  // 5th limit
+  // 6th fullCount
+  // 7th queryDepth
+
+  // TODO: This test is currently broken.
+  // Waiting for fake query server merge
+
+  GIVEN("there are no rows upstream") {
+    LimitExecutorInfos infos(0, 1, {}, 0, 1, true, 0);
```

Comment:
How can the input have 0 registers if the rows are copied into output that is 1 register wide?